core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -223,12 +289,16 @@ public void close() throws IOException {
 
         @Override
         public CloseableIterator<ScanTask> iterator() {
+          Expression partitionFilter =
+              Projections.inclusive(spec, isCaseSensitive()).project(baseTableFilter);
```

Comment:
You could have cached this and used `ManifestEvaluator.forPartitionFilter` but it is probably not worth it.